pyroomacoustics/bss/common.py

```diff
@@ -64,3 +64,102 @@ def projection_back(Y, ref, clip_up=None, clip_down=None):
         c[I] *= clip_down / np.abs(c[I])
 
     return c
+
+def sparir(G, S, weights=np.array([]), gini=0):
+    '''
+    2018 (c) Yaron Dibner, Virgile Hernicot, Juan Azcarreta MIT License
+
+   Fast proximal algorithm implementation for sparse approximation of (relative) impulse response from
+   incomplete measurements of the corresponding (relative) transfer function based on
+
+    Janský, Jakub & Koldovský, Zbyněk & Ono, Nobutaka. (2016). A computationally cheaper method for blind speech
+    separation based on AuxIVA and incomplete demixing transform. 1-5. IWAENC2016
+
+    Parameters
+    ----------
+    G: ndarray (nfrequencies, 1)
+        Frequency representation of the (incomplete) transfer function
+    S: ndarray (kfrequencies)
+        Indexes  of active frequency bins for sparse AuxIVA
+    weights: ndarray (kfrequencies) or int, optional
+        The higher the value of weights(i), the higher the probability that g(i)
+        is zero; if scalar, all weights are the same; if empty, default value is
+        used
+    gini: ndarray (nfrequencies)
+        Initialization for the computation of g
+
+    Returns
+    -------
+    Returns the sparse approximation of the impulse response in the
+    time-domain (real-valued) as an (nfrequencies) array.
+    '''
+
+    n_freq = G.shape[0]
+
+    y = np.concatenate((np.real(G[S]), np.imag(G[S])), axis=0)
+    M = y.shape[0]
+
+    if gini == 0:  # if no initialization is given
+        g = np.zeros((n_freq, 1))
+        g[0] = 1
+    else:
+        g = gini
+
+    if weights.size == 0:
+        tau = np.sqrt(n_freq) / (y.conj().T.dot(y))
+        tau = tau * np.exp(0.11 * np.abs((np.arange(1., n_freq + 1.).T)) ** 0.3)
+        tau = tau.T
+    elif weights.shape[0] == 1:
+        tau = np.ones((n_freq, 1)) * weights
+    else:
+        tau = np.tile(weights.T, (1, 1)).reshape(n_freq)
+
+    def soft(x, T):
+        if np.sum(np.abs(T).flatten()) == 0:
+            u = x
+        else:
+            u = np.max(np.abs(x) - T, 0)
+            u = u / (u + T) * x
+        return u
+
+    maxiter = 50    # maximum number of iterations to achieve convergence
+    alphamax = 1e5  # maximum step - length parameter alpha
+    alphamin = 1e-7  # minimum step - length parameter alpha
+    tol = 10
```

Comment:
I think these magic numbers should be replaced by optional parameters to the function with default values. Also make sure the purpose of these parameters is documented in the docstring.